applications/RANSApplication/custom_processes/rans_compute_reactions_process.cpp

```diff
@@ -0,0 +1,212 @@
+    Parameters rParameters)
+    : mrModel(rModel)
+{
+    KRATOS_TRY
+
+    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+    mEchoLevel = rParameters["echo_level"].GetInt();
+    mModelPartName = rParameters["model_part_name"].GetString();
+    mPeriodic = rParameters["consider_periodic"].GetBool();
+
+    KRATOS_CATCH("");
+}
+
+void RansComputeReactionsProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+
+    if (mPeriodic)
+    {
+        this->CorrectPeriodicNodes(mrModel.GetModelPart(mModelPartName), NORMAL);
+    }
+
+    KRATOS_CATCH("");
+}
+
+void RansComputeReactionsProcess::CorrectPeriodicNodes(
+    ModelPart& rModelPart,
+    const Variable<array_1d<double, 3>>& rVariable)
+{
+    KRATOS_TRY
+
+    auto& r_nodes = rModelPart.Nodes();
+
+    BlockPartition<ModelPart::NodesContainerType>(r_nodes).for_each([&](ModelPart::NodeType& rNode) {
+        if (rNode.Is(PERIODIC)) {
+            const int slave_id = rNode.FastGetSolutionStepValue(PATCH_INDEX);
+            const int master_id = rNode.Id();
+            if (master_id < slave_id) {
+                auto& master_value = rNode.FastGetSolutionStepValue(rVariable);
+
+                auto& slave_node = rModelPart.GetNode(slave_id);
+                auto& slave_value = slave_node.FastGetSolutionStepValue(rVariable);
+
+                const double master_value_norm = norm_2(master_value);
+                const double slave_value_norm = norm_2(slave_value);
+                const double value_norm = master_value_norm + slave_value_norm;
+
+                if (master_value_norm > 0.0) {
+                    noalias(master_value) = master_value * (value_norm / master_value_norm);
+                }
+
+                if (slave_value_norm > 0.0) {
+                    noalias(slave_value) = slave_value * (value_norm / slave_value_norm);
+                }
+            }
+        }
+    });
+
+    rModelPart.GetCommunicator().SynchronizeVariable(rVariable);
+
+    KRATOS_CATCH("");
+}
+
+void RansComputeReactionsProcess::ExecuteFinalizeSolutionStep()
+{
+    KRATOS_TRY
+
+    auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+    auto& r_nodes = r_model_part.Nodes();
+    VariableUtils().SetHistoricalVariableToZero(REACTION, r_nodes);
+
+    auto& r_conditions = r_model_part.Conditions();
+
+    BlockPartition<ModelPart::ConditionsContainerType>(r_conditions)
+        .for_each([&](ModelPart::ConditionType& rCondition) {
+            CalculateReactionValues(rCondition);
+        });
+
+    r_model_part.GetCommunicator().AssembleCurrentData(REACTION);
+    this->CorrectPeriodicNodes(r_model_part, REACTION);
+
+    BlockPartition<ModelPart::NodesContainerType>(r_nodes).for_each(
+        [&](ModelPart::NodeType& rNode) {
+            const auto& pressure_force =
+                rNode.FastGetSolutionStepValue(NORMAL) *
+                (-1.0 * rNode.FastGetSolutionStepValue(PRESSURE));
+            rNode.FastGetSolutionStepValue(REACTION) += pressure_force;
```

Comment:
```suggestion
            const auto& pressure_force = rNode.FastGetSolutionStepValue(NORMAL) * rNode.FastGetSolutionStepValue(PRESSURE);
            rNode.FastGetSolutionStepValue(REACTION) -= pressure_force;
```
Shouldn't you add the tangential component of the `REACTION` in here?

Reply:
OK. Now I see (it's added above).